packages/mobile/src/escrow/EscrowedPaymentListItem.tsx

```diff
@@ -13,33 +13,56 @@ import { EscrowedPayment } from 'src/escrow/actions'
 import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
 import { Namespaces, withTranslation } from 'src/i18n'
 import { inviteFriendsIcon } from 'src/images/Images'
+import { InviteDetails } from 'src/invite/actions'
+import { generateInviteLink, sendSms } from 'src/invite/saga'
 import { navigate } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
+import { RootState } from 'src/redux/reducers'
 import { divideByWei } from 'src/utils/formatting'
-import { navigateToURI } from 'src/utils/linking'
 import Logger from 'src/utils/Logger'
 
 interface OwnProps {
   payment: EscrowedPayment
 }
 
-type Props = OwnProps & WithTranslation
+interface StateProps {
+  invitees: InviteDetails[]
+}
+
+const mapStateToProps = (state: RootState): StateProps => ({
+  invitees: state.invite.invitees,
+})
+
+type Props = OwnProps & WithTranslation & StateProps
 
 const TAG = 'EscrowedPaymentListItem'
 
 export class EscrowedPaymentListItem extends React.PureComponent<Props> {
-  onRemind = () => {
-    const { payment, t } = this.props
+  onRemind = async () => {
+    const { payment, t, invitees } = this.props
     const recipientPhoneNumber = payment.recipientPhone
     CeloAnalytics.track(CustomEventNames.clicked_escrowed_payment_send_message)
-    // TODO: open up whatsapp/text message slider with pre populated message
+    // TODO(Tarik): add a UI that allows user to choose between SMS and Whatsapp (currently only SMS) for reminder message
     try {
-      if (Platform.OS === 'android') {
-        SendIntentAndroid.sendSms(recipientPhoneNumber, t('walletFlow5:escrowedPaymentReminderSms'))
+      const inviteDetails = invitees.find(
+        (inviteeObj) => recipientPhoneNumber === inviteeObj.e164Number
+      )
+      // OPEN QUESTION: Appears to me we won't have inivtee data when local data is wiped. is this an appropriate compromise in that case?
+      let message
+      if (!inviteDetails) {
+        message = t('walletFlow5:escrowedPaymentReminderSmsNoData')
       } else {
-        // TODO look into using MFMessageComposeViewController to prefill the body for iOS
-        navigateToURI(`sms:${recipientPhoneNumber}`)
+        const { inviteCode } = inviteDetails
+        // OPEN QUESTION: this function creates a unique link every time a reminder is sent. is this desirable?
+        const link = await generateInviteLink(inviteCode)
+        message = t('walletFlow5:escrowedPaymentReminderSms', {
+          // OPEN QUESTION: why do we have to send inviteCode and link if link is derived from inviteCode?
```

Comment:
Believe this is so someone can copy and paste the text message if the link doesn't work

Reply:
Got it, thanks.